tests/functional/conftest.py

```diff
@@ -191,6 +191,33 @@ def _wait(timeout=30, step=0.5):
     return _wait
 
 
+def wait_for_deleted(
+    *,
+    pg_manager: gitlab.base.RESTManager,
+    object_id: int,
+    description: str,
```

Comment:
I think we could potentially just take the description from `object.__name__`, that way we also explicitly know we're deleting via `ProjectMergeRequest` not `MergeRequest` etc.

Reply:
Sorry, I meant `object.__class__.__name__` here, purely to avoid having to define a description argument manually every time because it's nearly identical. Adding a description manually is a bit error prone to typos, e.g. as in https://github.com/python-gitlab/python-gitlab/pull/1784/files#diff-9810a028f51682f62bc400e8654365eca614e1905cc524507818b9a22329dca8R366.

Might also give us back those 1-line function signatures :grin:

Reply:
Done. I think. Only passing the `object` now.